Cow fortune styles

fairlighteth opened the pull request:
# Summary

- UI iteration on @shoom3301 's PR for the CoW fortune:


https://user-images.githubusercontent.com/31534717/233691501-a9321190-a5fc-4ab7-9501-c67059b3527d.mov


https://user-images.githubusercontent.com/31534717/233691598-83aedceb-8d2e-4184-911f-c74a32d04937.mov

@elena-zh Pending todo's:
- Need to check where the CoW fortune button (image) is undesired to be shown. E.g. on mobile when the wallet selector modal is open, in my opinion it's not desired to show the fortune cookie there. For mobile I'd say only show it where the footer toolbar is shown. For desktop I will further review.

elena-zh commented:
Hey @fairlighteth , 

> @elena-zh Pending todo's:
> 
> Need to check where the CoW fortune button (image) is undesired to be shown. E.g. on mobile when the wallet selector modal is open, in my opinion it's not desired to show the fortune cookie there. For mobile I'd say only show it where the footer toolbar is shown. For desktop I will further review.

Yes, it will be great to show the cookie on the footer toolbar, and not to show it above modals.

shoom3301 commented on src/cow-react/modules/fortune/containers/FortuneWidget/index.tsx:
```diff
@@ -1,156 +1,321 @@
         </FortuneBanner>
       )}
-      {isFortuneButtonVisible && <FortuneButton onClick={openRandomFortune}></FortuneButton>}
+      <FortuneButton onClick={openFortuneModal}>{isDailyFortuneChecked ? '' : '*'}</FortuneButton>
```
@fairlighteth can you add please styles for the case, when the daily fortune is not checked? * - is displayed

shoom3301 replied:
UPD:
https://github.com/cowprotocol/cowswap/pull/2361/commits/b3f6301b5c1cef4e3ed4cded7f07300b5735bf19
I disabled blinking animation when a daily fortune is checked

shoom3301 commented on src/cow-react/modules/fortune/containers/FortuneWidget/index.tsx:
```diff
@@ -1,156 +1,321 @@
   position: relative;
-  background: rgba(255, 77, 166, 0.09);
+  background: ${({ theme }) => theme.grey1};
+  opacity: ${({ isNewFortuneOpen }) => (isNewFortuneOpen ? 1 : 0.5)};
```
@fairlighteth we also need styles, when a fortune is open not for the first time (isNewFortuneOpen === false)

elena-zh commented:
Hey @shoom3301 , @fairlighteth , some more suggestions from this [thread](https://cowservices.slack.com/archives/C04KKRF42LU/p1682414770478169?thread_ts=1682335637.240619&cid=C04KKRF42LU): 
 - maybe instead of hiding the cookie forever by clicking the box we could hide it until the next day?
 - and hide the cookie when it is shared in Twitter (within the day). 

WDYT?

elena-zh commented:
Hey @fairlighteth , 
after a brief discussion with @shoom3301 we decided: 
1. On the 1st cookie opening:
 - show confettii
 - show the cookie
 - share on Twitter button
 - and close icon
2. If a user presses on the Share on Twitter button:
 - don't auto-close the cookie (a user will close it manually)
 - the cookie will no longer be displayed (until the next day)
3. If a user closes the cookie without charing on the twitter
 - the cookie will remain to be displayed on a page
 4. If user opens the cookie the 2nd time:
  - show an additional message 'Come back tomorrow for your daily fortune'
  - show the box 'I don't want to see this fortune again'
5. If a user checks the box 'I don't want to see this fortune again':
 - the cookie will no longer be displayed until the next cookie arrives
6. If a user presses on the Share on Twitter button (2nd opening):
 - see p.2

fairlighteth commented:
Pushed some updates:
- Changed title of the modal on second visit: `<>Already seen today's fortune? <br/> Return tomorrow for a fresh one!</>`
- Change CoW fortune cookie image to have a white outline (stands out more). Also added a box-shadow to have it stand out against the background.
- Modified checkbox label text: `<span>Hide today's fortune cookie</span>`
- Modified tweet text to say emphasize its your CoW fortune cookie and added the @CoWSwap twitter handle (so we get tagged and get notifications each time Tweets a fortune cookie!) --> `My CoW fortune cookie🐮: “${openFortune.text}” \n\n Get yours at swap.cow.fi @CoWSwap`

<img width="759" alt="Screenshot 2023-05-08 at 15 55 01" src="https://user-images.githubusercontent.com/31534717/236858030-c69e292c-fa81-4958-826a-6340657967a8.png">

<img width="581" alt="Screenshot 2023-05-08 at 15 54 25" src="https://user-images.githubusercontent.com/31534717/236858039-2e5ae2a7-5be5-468e-87fb-bd5a5a92c703.png">

https://user-images.githubusercontent.com/31534717/236858044-26e33c5a-2b61-4211-b845-35adc354904b.mov


https://user-images.githubusercontent.com/31534717/236858072-cccbf51b-8db9-4755-a9ac-46afe8b574b3.mov


https://user-images.githubusercontent.com/31534717/236858089-74abe599-8d55-4d5c-bcaa-f3c5ee80e9e1.mov

anxolin commented on src/custom/assets/cow-swap/Ableton Folder Info/dc66a3fa-0fe1-5352-91cf-3ec237e9ee90.xmp:
```diff
@@ -0,0 +1,17 @@
+<x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="XMP Core 5.6.0">
```
what´s this file??

fairlighteth replied:
Not sure how it got there and when. But removed in https://github.com/cowprotocol/cowswap/pull/2439

anxolin commented on src/cow-react/modules/fortune/hooks/useOpenRandomFortune.ts:
```diff
@@ -2,26 +2,35 @@ import { useAtom, useAtomValue } from 'jotai'
-    // TODO: What to do when there are no fortunes anymore?
-    const index = getRandomInt(0, list.length - 1)
-    const fortune = list[index]
+    let fortune = getRandomFortuneFromList(fortunesList.data, checkedFortunes)
```
i can't believe you used a `let` here 👀

kidding!

anxolin commented on public/cowsay.html:
```diff
@@ -0,0 +1,18 @@
+    />
+  </head>
+  <body></body>
+</html>
```
what is this HTML? shouldn't be in the PR description?

fairlighteth replied:
Removed in https://github.com/cowprotocol/cowswap/pull/2439 as I didn't see it being used. @shoom3301 please let me know in case I miss anything here.

shoom3301 replied:
Thanks for deleting this! It was a test for a link preview in Twitter

anxolin reviewed (commented):
Soft approve, i just don't get some files that are PR 

what are they for?

anxolin reviewed (approved):
Thanks for addressing the comments @fairlighteth , happy to consolidate this. 

Let us know if you need help with that
